### ISSUE TITLE
Question about Rate Limits

### ISSUE THREAD

**Issue by Tobias3107** [BenFradet/RiotSharp#617]
Hey, I have the Problem that i want do MultiThreading but if i do more then 5 Request it will send me a "Rate limit exceeded". So i have a Question will tjhe Rate Limits checked by RiotSharp. So that i can send so many requestion and the RiotSharp will wait for the RateLimits?

**Comment by BenFradet** [comment]
Yes riotsharp checks for rate limits and throttle requests, however I don't know if that still is the case when leveraging multithreading.

**Comment by Tobias3107** [comment]
It Should then i thing. Because it is the Same instance or iam false?

**Comment by BenFradet** [comment]
if you share the same instance across your threads yes but I don't know how thread safe our code is.

**Comment by kirides** [comment]
i had to use time-based rate limiting lately
i can totally recommend http://david-desmaisons.github.io/RateLimiter/ (currently at v2.1.0) for time based constraints. You can easily create time based, thread-safe limiters that work well with async-code (you can `await` a limiter)
